pkgs/by-name/br/bruno/package.nix

```diff
@@ -27,20 +27,22 @@ let
 in
 buildNpmPackage' rec {
   pname = "bruno";
-  version = "1.11.0";
+  version = "1.12.2";
 
   src = fetchFromGitHub {
     owner = "usebruno";
     repo = "bruno";
     rev = "v${version}";
-    hash = "sha256-Urskhzs00OEucoR17NDXNtnrcXk9h75E806Re0HvYyw=";
+    hash = "sha256-DJFZk3yPg2YYqTy4rXXZ18slI2e3+pLk2Tk0UHH3MHk=";
 
     postFetch = ''
+      patch -d $out <${./bruno-1-12-lockfile.patch}
```

Comment:
Each patch in nixpkgs applied to the upstream source should be documented. Out-of-tree patches, fetched using `fetchpatch`, are preferred.

In each patch comment, please explain the purpose of the patch and link to the relevant upstream issue if possible. If the patch has been merged upstream but is not yet part of the released version, please note the version number or date in the comment such that a future maintainer updating the nix expression will know whether the patch has been incorporated upstream and can thus be removed from nixpkgs.

Furthermore, please use a _stable_ URL for the patch. Rather than, for example, linking to a GitHub pull request of the form `https://github.com/owner/repo/pull/pr_number.patch`, which would change every time a commit is added or the PR is force-pushed, link to a specific commit patch in the form `https://github.com/owner/repo/commit/sha.patch`.

Here are two good examples of patch comments:

```nix
mkDerivation {
  …

  patches = [
    # Ensure RStudio compiles against R 4.0.0.
    # Should be removed next 1.2.X RStudio update or possibly 1.3.X.
    (fetchpatch {
      url = "https://github.com/rstudio/rstudio/commit/3fb2397c2f208bb8ace0bbaf269481ccb96b5b20.patch";
      sha256 = "0qpgjy6aash0fc0xbns42cwpj3nsw49nkbzwyq8az01xwg81g0f3";
    })
  ];
}
```

```nix
mkDerivation {
  …

  patches = [
    # Allow building with bison 3.7
    # PR at https://github.com/GoldenCheetah/GoldenCheetah/pull/3590
    (fetchpatch {
      url = "https://github.com/GoldenCheetah/GoldenCheetah/commit/e1f42f8b3340eb4695ad73be764332e75b7bce90.patch";
      sha256 = "1h0y9vfji5jngqcpzxna5nnawxs77i1lrj44w8a72j0ah0sznivb";
    })
  ];
}
```

Reply:
I'd remove that patch. `npm-lockfile-fix` should handle this issue.

Reply:
It didn't in this case. I don't know much about `npm-lockfile-fix`, but I've tried this same build with and without the patch; the build fails with the classic package not cached error without the path. It's the same error you can see in the logs from the auto-update attempt by r-ryantm.

Reply:
I'll fix this patch to match nixpkgs conventions but then we can decide if we want to merge this PR or not; the choices being: merge this PR which updates to 1.12.2 but guarantees we will need to manually remove the patch after the next version is released or alternatively holding off on updating Bruno in nixpkgs in hopes that 1.12.3 follows quickly with no need for the patch.

Reply:
Just add a comment on top of the patch explaining that this patch is only for version 1.12.2 and that it needs to be removed at the next point release and that will be enough.

Reply:
Fine fine.